debezium-core/src/main/java/io/debezium/transforms/outbox/EventRouter.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms.outbox;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.data.Envelope;
+import io.debezium.transforms.outbox.EventRouterConfigDefinition.AdditionalField;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.header.Headers;
+import org.apache.kafka.connect.transforms.ExtractField;
+import org.apache.kafka.connect.transforms.RegexRouter;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.debezium.transforms.outbox.EventRouterConfigDefinition.parseAdditionalFieldsConfig;
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+/**
+ * Debezium Outbox Transform Event Router
+ *
+ * @author Renato mefi (gh@mefi.in)
+ */
+public class EventRouter<R extends ConnectRecord<R>> implements Transformation<R> {
```

Comment:
@jpechane, can you add the new `@Incubating` annotation before mergin?

Reply:
I've rebased with master and added the annotation! :)